# - Replace the OpenFOAM third-party libraries with EASYBUILD variables.
# - Set install dir, compiler, etc using EASYBUILD
#
# authors: Ward Poelmans <wpoely86@gmail.com>, Kenneth Hoste (HPC-UGent), Mark Olesen <Mark.Olesen@esi-group.com>,
#          Simon Branford (University of Birmingham)
--- OpenFOAM-v1912/etc/bashrc.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/bashrc	2020-01-06 19:13:09.39069861 +0000
@@ -77,7 +77,7 @@
 # = SYSTEMOPENMPI | OPENMPI | SYSTEMMPI | MPI | MPICH | MPICH-GM |
 #   HPMPI | CRAY-MPICH | FJMPI | QSMPI | SGIMPI | INTELMPI | USERMPI
 # Also possible to use INTELMPI-xyz etc and define your own wmake rule
-export WM_MPLIB=SYSTEMOPENMPI
+export WM_MPLIB=EASYBUILDMPI
 
 
 #------------------------------------------------------------------------------
--- OpenFOAM-v1912/etc/config.sh/CGAL.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/config.sh/CGAL	2020-01-06 19:13:58.46916514 +0000
@@ -44,33 +44,9 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-boost_version=boost_1_64_0
-cgal_version=CGAL-4.9.1
-
-export BOOST_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$boost_version
-export CGAL_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$cgal_version
+##EasyBuild## settings
+export BOOST_ARCH_PATH="$EBROOTBOOST"
+export CGAL_ARCH_PATH="$EBROOTCGAL"
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
-if [ -n "$FOAM_VERBOSE" ] && [ -n "$PS1" ]
-then
-    echo "Using boost ($boost_version)  ->  $BOOST_ARCH_PATH" 1>&2
-    echo "Using CGAL ($cgal_version)  ->  $CGAL_ARCH_PATH" 1>&2
-fi
-
-if command -v _foamAddLibAuto >/dev/null  # Normal sourcing (not makeCGAL)
-then
-    _foamAddLibAuto $BOOST_ARCH_PATH  lib$WM_COMPILER_LIB_ARCH
-    _foamAddLibAuto $CGAL_ARCH_PATH   lib$WM_COMPILER_LIB_ARCH
-
-    # GMP/MPFR may have already been added with ThirdParty compiler, but cannot
-    # be certain so add here. Any duplicates will be removed later.
-
-    _foamAddLibAuto $GMP_ARCH_PATH   # No fallback libdir
-    _foamAddLibAuto $MPFR_ARCH_PATH  # No fallback libdir
-
-    unset boost_version cgal_version
-
-fi
-
-#------------------------------------------------------------------------------
--- OpenFOAM-v1912/etc/config.sh/FFTW.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/config.sh/FFTW	2020-01-06 19:14:18.89030368 +0000
@@ -35,23 +35,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-fftw_version=fftw-3.3.7
-export FFTW_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$fftw_version
+##EasyBuild## settings
+export FFTW_ARCH_PATH="$EBROOTFFTW"
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
-if [ -n "$FOAM_VERBOSE" ] && [ -n "$PS1" ]
-then
-    echo "Using fftw ($fftw_version)  ->  $FFTW_ARCH_PATH" 1>&2
-fi
-
-if command -v _foamAddLibAuto >/dev/null  # Normal sourcing (not makeFFTW)
-then
-
-    _foamAddLibAuto $FFTW_ARCH_PATH  lib$WM_COMPILER_LIB_ARCH
-
-    unset fftw_version
-
-fi
-
-#------------------------------------------------------------------------------
--- OpenFOAM-v1912/etc/config.sh/gperftools.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/config.sh/gperftools	2020-01-06 19:14:52.90383440 +0000
@@ -38,32 +38,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-gperftools_version=gperftools-2.5
-GPERFTOOLS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$gperftools_version
+##EasyBuild## settings
+GPERFTOOLS_ARCH_PATH="$EBROOTGPERFTOOLS"
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
-if [ -n "$FOAM_VERBOSE" ] && [ -n "$PS1" ]
-then
-    echo "Using gperftools ($gperftools_version)  ->  $GPERFTOOLS_ARCH_PATH" 1>&2
-fi
-
-if command -v _foamAddLib >/dev/null  # Normal sourcing
-then
-
-    # If GPERFTOOLS_ARCH_PATH does not end with '-system' or '-none',
-    # it is located within ThirdParty, or a central installation
-    # outside of ThirdParty and must be added to the lib-path.
-
-    ending="${GPERFTOOLS_ARCH_PATH##*-}"
-    if [ "$ending" != none ] && [ "$ending" != system ]
-    then
-        _foamAddLib     $GPERFTOOLS_ARCH_PATH/lib$WM_COMPILER_LIB_ARCH
-        _foamAddPath    $GPERFTOOLS_ARCH_PATH/bin
-    fi
-
-    unset gperftools_version ending
-
-fi
-
-#------------------------------------------------------------------------------
--- OpenFOAM-v1912/etc/config.sh/metis.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/config.sh/metis	2020-01-06 19:14:21.70845195 +0000
@@ -34,8 +34,9 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-METIS_VERSION=metis-5.1.0
-export METIS_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$METIS_VERSION
+##Easybuild## settings
+METIS_VERSION="metis-$EBVERSIONMETIS"
+[ -d "$METIS_ARCH_PATH" ] || METIS_ARCH_PATH="$METIS_ROOT"
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
--- OpenFOAM-v1912/etc/config.sh/mpi.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/config.sh/mpi	2020-01-06 19:12:41.54940690 +0000
@@ -96,6 +96,10 @@
     _foamEtc -config  mpi-user                  # <- Adjustments (optional)
     ;;
 
+EASYBUILDMPI)
+    export FOAM_MPI=mpi
+    ;;
+
 SYSTEMMPI)
     export FOAM_MPI=mpi-system
     _foamEtc -config  mpi-system                # <- Adjustments (optional)
--- OpenFOAM-v1912/etc/config.sh/paraview.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/config.sh/paraview	2020-01-06 19:12:41.66878632 +0000
@@ -54,7 +54,8 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-ParaView_VERSION=5.6.3
+##Easybuild## settings
+ParaView_VERSION="$EBVERSIONPARAVIEW"
 ParaView_QT=qt-system
 cmake_version=cmake-system
 
@@ -85,14 +86,6 @@
     esac
 done
 
-# ThirdParty cmake
-cmake="$archDir/$cmake_version"
-if [ -r "$cmake/bin/cmake" ]
-then
-    # _foamAddPath not available when foamPV function is used
-    PATH="$cmake/bin:$PATH"
-fi
-
 case "$ParaView_VERSION" in
 ('')
     # empty - do nothing
@@ -129,8 +130,9 @@
         ;;
     esac
 
-    export ParaView_DIR="$archDir/ParaView-$ParaView_VERSION"
+    ##Easybuild## settings
+    export ParaView_DIR="$EBROOTPARAVIEW"
     unset PV_PLUGIN_PATH
 
     # Set paths if binaries are present
     if [ -r "$ParaView_DIR" ]
@@ -133,26 +142,6 @@
         export PV_PLUGIN_PATH="$FOAM_LIBBIN/paraview-$pv_api"
         export PATH="$ParaView_DIR/bin:$PATH"
 
-        # QT libraries as required, and Qt5_DIR for the root directory.
-        # Another possibility: "qtpaths --qt-version"
-        qtDir="$archDir/$ParaView_QT"
-        if [ -d "$qtDir" ]
-        then
-            case "$ParaView_QT" in
-            (*-5*)
-                export Qt5_DIR="$qtDir"
-                ;;
-            esac
-            for libDir in "lib$WM_COMPILER_LIB_ARCH" "lib"
-            do
-                if [ -d "$qtDir/$libDir" ]
-                then
-                    export LD_LIBRARY_PATH="$qtDir/$libDir:$LD_LIBRARY_PATH"
-                    break
-                fi
-            done
-        fi
-
         # ParaView libraries
         # - 5.4 and earlier: lib/paraview-X.X
         # - 5.5 and later:   lib/, but could also be lib64/
--- OpenFOAM-v1912/etc/config.sh/scotch.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/config.sh/scotch	2020-01-06 19:14:50.81742938 +0000
@@ -34,8 +34,10 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-SCOTCH_VERSION=scotch_6.0.9
-export SCOTCH_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$SCOTCH_VERSION
+##Easybuild## settings
+export SCOTCH_VERSION="scotch_$EBVERSIONSCOTCH"
+export SCOTCH_ARCH_PATH="$EBROOTSCOTCH"
+[ -d "$SCOTCH_ARCH_PATH" ] || SCOTCH_ARCH_PATH="$SCOTCH_ROOT"
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
--- OpenFOAM-v1912/etc/config.sh/vtk.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/etc/config.sh/vtk	2020-01-06 19:13:23.08425391 +0000
@@ -34,53 +34,12 @@
 #------------------------------------------------------------------------------
 # USER EDITABLE PART: Changes made here may be lost with the next upgrade
 
-vtk_version=VTK-8.2.0
-mesa_version=mesa-17.1.1
-mesa_llvm=none
-
-export VTK_DIR="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$vtk_version"
-export MESA_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER/$mesa_version"
-export LLVM_ARCH_PATH="$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH/$mesa_llvm"
+##Easybuild## settings
+export VTK_DIR="$EBROOTVTK"
+export MESA_ARCH_PATH="$EBROOTMESA"
 
 # Define paraview-mesa directory as required
 unset ParaView_MESA_DIR
 
 # END OF (NORMAL) USER EDITABLE PART
 #------------------------------------------------------------------------------
-if [ -n "$FOAM_VERBOSE" ] && [ -n "$PS1" ]
-then
-    echo "Using vtk" 1>&2
-    echo "    vtk  ($vtk_version) $VTK_DIR" 1>&2
-    echo "    mesa ($mesa_version) $MESA_ARCH_PATH" 1>&2
-    echo "    llvm ($mesa_llvm) $LLVM_ARCH_PATH" 1>&2
-    if [ -d "$ParaView_MESA_DIR" ]
-    then
-    echo "    paraview-mesa $ParaView_MESA_DIR" 1>&2
-    fi
-fi
-
-if command -v _foamAddLib >/dev/null  # normal sourcing
-then
-
-    # Set paths if binaries are present
-    if [ -d "$VTK_DIR" ]
-    then
-        # Uses lib/ and not lib64/
-        _foamAddLib "$VTK_DIR/lib"
-    else
-        unset VTK_DIR
-    fi
-
-    if [ -d "$MESA_ARCH_PATH" ]
-    then
-        _foamAddLibAuto "$LLVM_ARCH_PATH"
-        _foamAddLibAuto "$MESA_ARCH_PATH"
-    else
-        unset LLVM_ARCH_PATH
-        unset MESA_ARCH_PATH
-    fi
-
-    unset vtk_version mesa_version mesa_llvm
-fi
-
-#------------------------------------------------------------------------------
--- OpenFOAM-v1912/wmake/scripts/have_readline.orig	2020-01-06 19:07:23.000000000 +0000
+++ OpenFOAM-v1912/wmake/scripts/have_readline	2020-01-06 19:15:54.77256611 +0000
@@ -58,7 +58,7 @@
     # warn="==> skip readline"
 
     # Expected location, include/library names
-    prefix=system
+    prefix="$EBROOTLIBREADLINE"
     incName="readline/readline.h"
     libName="libreadline"
 
